qiskit_dynamics/pulse/pulse_to_signals.py

```diff
@@ -163,9 +163,9 @@ def get_signals(self, schedule: Schedule) -> List[DiscreteSignal]:
             )
 
         for start_sample, inst in schedule.instructions:
-            chan = inst.channel.name
-            phi = phases[chan]
-            freq = frequency_shifts[chan]
+            # get channel name if instruction has it
+            chan = getattr(inst, "channel", None)
```

Comment:
oops, didn't see the name in there

```suggestion
            chan = inst.channel.name if hasattr(inst, "channel") else None
```